common/utils.go

```diff
@@ -46,13 +47,17 @@ func TransactionLogging(reducedLoggingResponseCodes []int, logger kitlog.Logger)
 	errorLogger := logging.Error(logger)
 	return func(ctx context.Context, code int, r *http.Request) {
 		tid, _ := ctx.Value(ContextKeyRequestTID).(string)
-		transactionInfoLogger, ok := ctx.Value(ContextKeyTransactionInfoLogger).(kitlog.Logger)
-		traceId, spanId := candlelight.ExtractTraceInformation(r.Context())
-
-
+		transactionInfoLogger, transactionLoggerOk := ctx.Value(ContextKeyTransactionInfoLogger).(kitlog.Logger)
+		traceId, spanId, traceOk := candlelight.ExtractTraceInformation(r.Context())
+		var traceLogKVs []interface{}
+		if traceOk {
+			traceLogKVs = append(traceLogKVs, candlelight.SpanIDLogKeyName, spanId, candlelight.TraceIdLogKeyName, traceId)
+		}
 
-		if !ok {
-			errorLogger.Log(logging.MessageKey(), "transaction logger not found in context", "tid", tid,candlelight.SpanIDLogKeyName, spanId,candlelight.TraceIdLogKeyName,traceId)
+		if !transactionLoggerOk {
```

Comment:
As a general theme, try to move this kind of conditional logic into the `candlight` package (see above)